### ISSUE TITLE
Tiny_tds fat binary gem (mingw-x64/ Windows) dependency questions

### ISSUE THREAD

**Issue by senolfeldmann** [rails-sqlserver/tiny_tds#500]
I have a question about tiny_tds. Does anyone know if the the various .exe binaries which are shipped in the fat binary gem version (mingw-x64 / CRuby on Windows) of tiny_tds are really needed for tiny_tds to be able to operate? My takeaway from reading the various rake files is that only some libs are needed.
For me, iconv.exe in C:\Ruby27-x64\lib\ruby\gems\2.7.0\gems\tiny_tds-2.1.5-x64-mingw32\ports\x86_64-w64-mingw32\libiconv\1.15\bin is especially important, as it is licensed under GPL and I want to use activerecord-sqlserver-adapter (and thus tiny_tds as it is a dependency) commercially.

And what about the .exe binaries of freetds: bsqldb.exe, datacopy.exe, defncopy.exe, freebcp.exe, tdspool.exe and tsql.exe, are they needed or can I delete them from my system?

**Comment by andyundso** [comment]
> Does anyone know if the the various .exe binaries which are shipped in the fat binary gem version (mingw-x64 / CRuby on Windows) of tiny_tds are really needed for tiny_tds to be able to operate?

No, these are needed to operate. However, if you report an issue, it is helpful for us if you run the `tsql` utility to give us more details about your environment.

> For me, iconv.exe in C:\Ruby27-x64\lib\ruby\gems\2.7.0\gems\tiny_tds-2.1.5-x64-mingw32\ports\x86_64-w64-mingw32\libiconv\1.15\bin is especially important, as it is licensed under GPL and I want to use activerecord-sqlserver-adapter (and thus tiny_tds as it is a dependency) commercially.

FreeTDS is also distributed under the [GPL license](https://github.com/FreeTDS/freetds/blob/master/COPYING.txt).

I am no expert on licensing. I think you are good if you would provide your software as a [SaaS](https://www.mend.io/blog/the-saas-loophole-in-gpl-open-source-licenses/). If you plan to distribute your software alongside a copy of `tiny_tds`, then I think you would have to license your code as `GPL` as well.

In theory, we do not have to use `libiconv` for `FreeTDS`. It would then fallback to its own `iconv`-like implementation, but again, it does not matter since `FreeTDS` is also licensed as GPL.

**Comment by andyundso** [comment]
@senolfeldmann I am not sure why `tiny_tds` is licensed with MIT, that was a decision taken probably more than a decade ago. based on the README, I would assume the `tiny_tds` code itself is licensed as MIT, but the GPL restrictions from FreeTDS and libiconv start applying once `tiny_tds` and its dependencies are shipped together.
